module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -2449,22 +2470,37 @@ protected function mapRenewalBlockReason($reason, $itype)
     }
 
     /**
-     * Return a location for a Koha item
+     * Return a location (branch or shelving) for a Koha item
      *
      * @param array $item Item
      *
      * @return string
      */
     protected function getItemLocationName($item)
     {
-        $libraryId = (!$this->useHomeLibrary && null !== $item['holding_library_id'])
-            ? $item['holding_library_id'] : $item['home_library_id'];
-        $name = $this->translateLocation($libraryId);
-        if ($name === $libraryId) {
-            $libraries = $this->getLibraries();
-            $name = isset($libraries[$libraryId])
-                ? $libraries[$libraryId]['name'] : $libraryId;
+        $name = '';
+
+        switch ($this->config['ItemTypeRenewalBlockMappings']['Location']) {
+            case 'Shelving':
+                $shelvingLocationId = $item['location'];
+                $name = $this->translateLocation($shelvingLocationId);
+                if ($name === $shelvingLocationId) {
+                    $shelvingLocations = $this->getShelvingLocations();
+                    $name = isset($shelvingLocations[$shelvingLocationId])
```

Comment:
I believe you could simplify this to:

```
$name = $shelvingLocations[$shelvingLocationId]['description'] ?? $shelvingLocationId;
```

That also protects against the (probably unlikely, but nothing is impossible) situation where the data is set by the ID but somehow missing the location element.

Reply:
(And I see now that this code is based on pre-existing code for processing library names below... but I still think it might be worth improving, and you could make a matching change to the name logic).

Reply:
thanks @demiankatz . Sorry I missed this change earlier. I've gone ahead and fixed this in my follow-up commit.